server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -30,21 +43,80 @@ class ObjectService(googleStorageService: GoogleStorageService[IO]) extends Http
       } yield resp
   }
 
-  def localize(req: Localize): IO[Unit] = ???
+  def localize(req: Localize): IO[Unit] = {
+    val res = Stream.emits(req.entries).map {
+      entry =>
+        googleStorageService.getObject(entry.bucketNameAndObjectName.bucketName, entry.bucketNameAndObjectName.blobName, None) //get file from google
+          .through(io.file.writeAll(Paths.get(entry.localObjectPath.asString), blockingEc)) //write file to local disk
+    }.parJoin(10)
+
+    res.compile.drain
+  }
 
-  def checkMeta(req: GetMetadataRequest): IO[Unit] = ???
+  def checkMetadata(req: GetMetadataRequest): IO[MetadataResponse] =
+    for {
+      traceId <- IO(TraceId(randomUUID()))
+      storageLinks <- readJsonFileToA[IO, List[StorageLink]](pathToStorageLinks).compile.lastOrError
+      storageLink = storageLinks.find(_.localBaseDirectory == req.localObjectPath)
```

Comment:
path + matching logic needs to be a bit more nuanced, i.e. to handle the `recursive` flag. May be fine for now but should be tracked somewhere.

Reply:
good point. I'll leave a TODO. I haven't had a chance to ask about what `recursive` flag does yet.